gnomad_constraint/utils/generic.py

```diff
@@ -0,0 +1,246 @@
+# noqa: D100
+from typing import List, Tuple, Optional, Union, Any
+
+import hail as hl
+
+# cSpell: disable
+def fast_filter_vep(
+    t: hl.Table,
+    vep_root: str = "vep",
+    syn: bool = True,
+    canonical: bool = True,
+    filter_empty: bool = True,
+) -> hl.Table:
+    """
+    Filter to variants where 'transcript_consequences' within the VEP annotation is not empty.
+
+    Also filter to variants where 'most_severe_consequence' is 'synonymous' and the transcript is the
+    canonical transcript, if 'syn' and 'canonical' parameter are set to True, respectively.
+
+    :param t: Input Table or MatrixTable.
+    :param vep_root: Name used for VEP annotation. Defaults to 'vep'.
+    :param syn: Whether to filter to variants where the most severe consequence is "synonymous". Defaults to True.
+    :param canonical: Whether to filter to only canonical transcripts. Defaults to True.
+    :param filter_empty: Whether to filter out rows where 'transcript_consequences' is empty. Defaults to True.
+    :return: Table or MatrixTable filtered to specified criteria.
+    """
+    transcript_csqs = t[vep_root].transcript_consequences
+    criteria = [lambda csq: True]
+    if syn:
+        criteria.append(lambda csq: csq.most_severe_consequence == "synonymous_variant")
+    if canonical:
+        criteria.append(lambda csq: csq.canonical == 1)
+    transcript_csqs = transcript_csqs.filter(lambda x: combine_functions(criteria, x))
```

Comment:
When will the lambda return `False`? I feels that `combine_functions` always return `True` and is not necessary